libfreerdp/core/info.c

```diff
@@ -671,35 +817,63 @@ BOOL rdp_recv_logon_info_v1(rdpRdp* rdp, wStream* s, logon_info *info)
 		return FALSE;
 
 	Stream_Read_UINT32(s, cbDomain); /* cbDomain (4 bytes) */
-	if (cbDomain > 52)
-		return FALSE;
-	if (cbDomain)
+
+	/* cbDomain is the size of the Unicode character data (including the mandatory
+	 * null terminator) in bytes present in the fixed-length (52 bytes) Domain field
+	 */
+	if ((cbDomain % 2) || cbDomain < 2 || cbDomain > 52)
 	{
-		ConvertFromUnicode(CP_UTF8, 0, (WCHAR*) Stream_Pointer(s), cbDomain, &info->domain, 0, NULL, FALSE);
```

Comment:
Same here I think we should not test the case cbDomain < 2 and consider this as "info not provided".